core/src/block.rs

```diff
@@ -529,7 +558,10 @@ mod valid {
             let topology = Topology::new(peers);
 
             let mut block = ValidBlock::new_dummy();
-            let payload = block.payload().clone();
+            let payload = {
+                let SignedBlock::V1(signed) = &block.0;
+                signed.payload.clone()
+            };
```

Comment:
this is often repeated. Make it into a function. It's ok to have `payload` function in this case

Reply:
Disagree. It repeats only in tests and only 4 times. I don't think it's a good idea to make a function only for testing

Reply:
@Erigara can you share your opinion here?

Reply:
> Disagree. It repeats only in tests and only 4 times. I don't think it's a good idea to make a function only for testing

I see your point now. How about you define a freestanding function in the `test` module instead of a method